commit cd544890ced4192d07467c89e23adbb62d8cea5c
Author: Milian Wolff <mail@milianw.de>
Date:   Mon Dec 18 11:40:35 2017 +0100

    Fix build with CMake 3.10
    
    Looks like a classic false-positive, but this makes the compile
    pass for me without making the code harder to read:
    
    AutoMoc error
    -------------
      "/ssd/milian/projects/kf5/src/kde/workspace/kwin/kcmkwin/kwinscripts/main.cpp"
    The file contains a K_PLUGIN_FACTORY macro, but does not include "main.moc"!
    Consider to
     - add #include "main.moc"
     - enable SKIP_AUTOMOC for this file
    
    So we just add the include and then get rid of the duplicate
    definition of the plugin factory and the problem is resolved.

diff --git a/kcmkwin/kwinscripts/main.cpp b/kcmkwin/kwinscripts/main.cpp
index f5ee04b30..baa517511 100644
--- a/kcmkwin/kwinscripts/main.cpp
+++ b/kcmkwin/kwinscripts/main.cpp
@@ -22,3 +22,5 @@
 
 K_PLUGIN_FACTORY(KcmKWinScriptsFactory,
                  registerPlugin<Module>("kwin-scripts");)
+
+#include "main.moc"
diff --git a/kcmkwin/kwinscripts/module.cpp b/kcmkwin/kwinscripts/module.cpp
index a0d698ec2..ccf7d4130 100644
--- a/kcmkwin/kwinscripts/module.cpp
+++ b/kcmkwin/kwinscripts/module.cpp
@@ -40,8 +40,6 @@
 
 #include "version.h"
 
-K_PLUGIN_FACTORY_DECLARATION(KcmKWinScriptsFactory)
-
 Module::Module(QWidget *parent, const QVariantList &args) :
     KCModule(parent, args),
     ui(new Ui::Module),
